Fix k3s cloud-provider docs (#254)
harvester/docs: docs/rancher/cloud-provider.md

### Deploying to the K3s Cluster with Harvester Node Driver [Experimental]

When spinning up a K3s cluster using the Harvester node driver, you can perform the following steps to deploy the harvester cloud provider:

> 1. Generate and inject cloud config for `harvester-cloud-provider`

Reviewer: ```suggestion
1. Generate and inject cloud config for `harvester-cloud-provider`.
```

> The cloud provider needs a kubeconfig file to work, a limited scoped one can be generated using the [generate_addon.sh](https://raw.githubusercontent.com/harvester/cloud-provider-harvester/master/deploy/generate_addon.sh) script available in the [harvester/cloud-provider-harvester](https://github.com/harvester/cloud-provider-harvester) repo.

Reviewer: ```suggestion
The cloud provider needs a kubeconfig file to work. A scope-limited one can be generated using the [generate_addon.sh](https://raw.githubusercontent.com/harvester/cloud-provider-harvester/master/deploy/generate_addon.sh) script available in the [harvester/cloud-provider-harvester](https://github.com/harvester/cloud-provider-harvester) repo.
```

Reviewer: Not sure if I understand `limited scoped` correctly. Please double check

:::note

> The script depends on `kubectl` and `jq` to operate the Harvester cluster

Reviewer: ```suggestion
The script depends on `kubectl` and `jq` to operate the Harvester cluster.
```

> The script needs access to the `Harvester Cluster` kubeconfig to work.

Reviewer: ```suggestion
The script needs access to the `Harvester Cluster` kubeconfig.
```

> The namespace needs to be the namespace in which the guest cluster will be created.

Reviewer: ```suggestion
The namespace needs to be the one in which the guest cluster will be created.
```

users:
- name: harvester-cloud-provider-default-local
  user:
    token: <TOKEN>
    
    
########## cloud-init user data ############
write_files:
- encoding: b64
  content: <CONTENT>
  owner: root:root
  path: /etc/kubernetes/cloud-config
  permissions: '0644'
```

> Copy and paste the output below `cloud-init user data` to **Machine Pools >Show Advanced > User Data**.

Reviewer: ```suggestion
Copy the output below `cloud-init user data` and paste it to **Machine Pools >Show Advanced > User Data**.
```

![](/img/v1.1/rancher/cloud-config-userdata.png)

> 3. Add the following `HelmChart` yaml of `harvester-cloud-provider` to **Cluster Configuration > Add-On Config > Additional Manifest**

Reviewer: ```suggestion
3. Add the following `HelmChart` yaml of `harvester-cloud-provider` to **Cluster Configuration > Add-On Config > Additional Manifest**.
```

> 3. Add the following `HelmChart` yaml of `harvester-cloud-provider` to **Cluster Configuration > Add-On Config > Additional Manifest**

Author: No, otherwise, the k3s cluster cannot be active and Apps cannot be used from the UI

```
apiVersion: helm.cattle.io/v1
kind: HelmChart
metadata:
  name: harvester-cloud-provider
  namespace: kube-system
spec:
  targetNamespace: kube-system
  bootstrap: true
  repo: https://charts.harvesterhci.io/
  chart: harvester-cloud-provider
  version: 0.1.13
  helmVersion: v3
```

![](/img/v1.1/rancher/external-cloud-provider-addon.png)

> 4. Disable the in-tree cloud provider by

Reviewer: ```suggestion
4. Disable the in-tree cloud provider as follows:
```

Reviewer: If we use `by` here, all the verbs below should be in the`ing` form

> - Click the `Edit as YAML` button

Reviewer: ```suggestion
- Click the `Edit as YAML` button.
```

![](/img/v1.1/rancher/edit-k3s-cluster-yaml.png)

> - Disable `servicelb` and Set `disable-cloud-controller: true` to disable default k3s cloud controller.

Reviewer: ```suggestion
- Disable `servicelb` and set `disable-cloud-controller: true` to disable the default k3s cloud controller.
```

```yaml
    machineGlobalConfig:
      disable:
        - servicelb
      disable-cloud-controller: true
```

> - Add `cloud-provider=external` to use harvester cloud provider.

Reviewer: ```suggestion
- Add `cloud-provider=external` to use the Harvester cloud provider.
```

```yaml
    machineSelectorConfig:
      - config:
          kubelet-arg:
          - cloud-provider=external
          protect-kernel-defaults: false
```

![](/img/v1.1/rancher/k3s-cluster-yaml-content-for-harvester-cloud-provider.png)

> With these settings in place a K3s cluster should provision successfully while using the external cloud provider.

Reviewer: ```suggestion
With these settings in place, a K3s cluster should be provisioned successfully using the external cloud provider.
```
